pkg/machines/libvirt-dbus.es6

```diff
@@ -149,23 +149,23 @@ LIBVIRT_DBUS_PROVIDER = {
 
     ATTACH_DISK({
         connectionName,
-        diskFileName,
+        poolName,
+        volumeName,
         target,
         vmId,
         vmName,
         permanent,
         hotplug
     }) {
+        const args = `--disk target=${target},source_pool=${poolName},source_volume=${volumeName}`;
```

Comment:
All of these arguments need to be shell-escaped, in case they contain characters like ` `, `$`, or `'`.

Instead of doing it for every invocation, I recommend passing the arguments as a dict or array of (option, value) pairs (in case the order matters) to `virtBuildXML()`, and escape them there.

Reply:
Bugzilla for libvirt-dbus: https://bugzilla.redhat.com/show_bug.cgi?id=1647823